### DISCUSSION TITLE
How to use with expo-constants on web?

### DISCUSSION THREAD

**Discussion by AllanChain** [goatandsheep/react-native-dotenv#274] [Q&A] [answered]
When using this plugin, I can't access properties of `Constants.manifest` on expo web. `Constants.manifest` is an empty object.

That's because expo web uses webpack `DefinedPlugin`, and relies on `process.env.APP_MANIFEST` (which is `undefined` after using this plugin) to get `Constants.manifest`.

Code:

`babel.config.js`:
```js
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: [
      ["module:react-native-dotenv"]
    ]
  };
};
```

`App.js`:
```js
import { Text, View } from 'react-native';
import Constants from 'expo-constants'

export default function App() {
  return (
    <View>
      <Text>Name: {Constants.manifest.name}</Text>
    </View>
  );
}
```

The page should show app name, but it's empty after using this plugin. Is there any way to configure it to work with `DefinePlugin`?

**Comment by AllanChain** [comment, marked as answer]
I just realized that I can put it in `overrides`:

```js
module.exports = function (api) {
  api.cache(true)
  return {
    presets: ['babel-preset-expo'],
    overrides: [
      {
        test: (fileName) => !fileName.includes('node_modules'),
        plugins: [
          ['module:react-native-dotenv']
        ]
      }
    ]
  }
}
```

**Reply by goatandsheep** [reply]
This i have to try...
